src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,56 +1,133 @@
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setimgUrl] = useState('');
+  const [imdbUrl, setimdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const handleTitleUpdate = (newValue: string) => {
+    setTitle(newValue);
+  };
+
+  const handleDescriptionUpdate = (newValue: string) => {
+    setDescription(newValue);
+  };
+
+  const handleimgUrlUpdate = (newValue: string) => {
+    setimgUrl(newValue);
+  };
+
+  const handleImdbUrlUpdate = (newValue: string) => {
+    setimdbUrl(newValue);
+  };
+
+  const handleImdbIdUpdate = (newValue: string) => {
+    setImdbId(newValue);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setDescription('');
+    setimgUrl('');
+    setimdbUrl('');
+    setImdbId('');
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
+
+    setCount(count + 1);
+
+    reset();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={handleTitleUpdate}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={description}
+        onChange={handleDescriptionUpdate}
       />
 
       <TextField
         name="imgUrl"
         label="Image URL"
-        value=""
+        value={imgUrl}
+        onChange={handleimgUrlUpdate}
+        required
       />
 
       <TextField
         name="imdbUrl"
         label="Imdb URL"
-        value=""
+        value={imdbUrl}
+        onChange={handleImdbUrlUpdate}
+        required
       />
 
       <TextField
         name="imdbId"
         label="Imdb ID"
-        value=""
+        value={imdbId}
+        onChange={handleImdbIdUpdate}
+        required
       />
 
       <div className="field is-grouped">
         <div className="control">
-          <button
-            type="submit"
-            data-cy="submit-button"
-            className="button is-link"
-          >
-            Add
-          </button>
+          {!title || !imgUrl || !imdbUrl || !imdbId ? (
+            <button
+              type="submit"
+              data-cy="submit-button"
+              className="button is-link"
+              disabled
```

Comment:
```suggestion
              disabled={condition}
```
And just use it here. No need to render two different buttons